src/util/compareThought.ts

```diff
@@ -221,3 +223,21 @@ export const compareThoughtByUpdated: ComparatorFunction<Thought> = (a: Thought,
 /** Compare two thoughts by their lastUpdated timestamp in descending order (newest first). Fall back to compareReasonable if created at the same time. */
 export const compareThoughtByUpdatedDescending: ComparatorFunction<Thought> = (a: Thought, b: Thought) =>
   compare(b.lastUpdated, a.lastUpdated) || compareReasonable(a.value, b.value)
+
+/** Get note value or fallback for a Thought or string input. */
+const getNoteOrFallback = (state: State, input: Thought | string): string => {
+  if (typeof input === 'string') return input
+  return noteValue(state, input.id) || input.rank.toString()
+}
+
+/** Compare two thoughts by their note value in ascending order, falling back to their rank if notes are absent or equal. */
+export const compareThoughtByNote =
+  (state: State): ComparatorFunction<Thought | string> =>
```

Comment:
`Thought | string` doesn't really make sense as a union type. Either it's a thought or a string, but never both. Perhaps this allows you to reuse certain functions by making them polymorphic, but I don't think it's the right kind of abstraction.

It would be better to extract the note value and then compare it using a string-only comparator.